app/models/checkout.rb

```diff
@@ -18,6 +18,25 @@ def status
     fields['status']
   end
 
+  def status_human
+    return unless overdue?
+
+    claims_returned? ? 'Claims Returned' : 'Overdue'
+  end
+
+  def due_date_human
+    due_dates = []
+
+    if recalled?
+      due_dates << 'Recalled'
+      due_dates << format_due_date(recall_due_date)
+    end
+
+    due_dates << format_due_date(due_date)
+
+    due_dates.join('<br>')
```

Comment:
```suggestion
  def due_date_human
    format_due_date due_date
  end
  
  def recalled_status
    'Recalled' if recalled?
  end
```

I think we should try to keep any markup out of our models in general, and I'm sort of additionally thinking we should just have this method return the human readable due date (the formatted due date). Then in the renewal job can call recalled_status and due_date_human separately.

Reply:
refactored this and moved that logic to a view component, pls let me know what you think about it?

Reply:
I was thinking simpler I think, something like this directly in the erb:

```rb
<td>
  <%= "Recalled #{@checkout.recall_due_date} <br>" if @checkout.recalled? %>
  <%= @checkout.due_date %>
</td>
```

I don't think you need to use a view component here because the logic isn't too bad, but maybe I'm missing something

Reply:
I remember why I had moved the logic to model in the first place, so I do not have to redo it in js when I get the new due date back from the job. It is among the fields that gets updated with renewal. View component is nice but not helping much in this case. I am thinking to move it back to model 🤔

Reply:
It is in the model though, right? You can render this small bit of ERB in the Job itself (server side render it) and store the results in Redis. Maybe we can look at this together over Zoom.

Reply:
not in the model anymore, I moved it out to the view component, newer version of the model here: https://github.com/psu-libraries/myaccount/pull/195/files#diff-2f2015ebe9d7db97c62891921e1d685c

Reply:
I am trying to find a way that I can use for display checkout and renew checkout

Reply:
I know about the view component.

Why can't you put [this](https://github.com/psu-libraries/myaccount/pull/195#discussion_r427724959) in a partial and render it in a job? If we are producing HTML it should be done by a view not a model is what I'm saying.

Reply:
I added the changes for  due date partial and used it for both renew and display as you suggested